pkg/core/proxy/util.go

```diff
@@ -16,3 +25,60 @@ func writeNsswitchConfig(logger *zap.Logger, nsSwitchConfig string, data []byte,
 	}
 	return nil
 }
+
+func (p *Proxy) globalPassThrough(ctx context.Context, client, dest net.Conn) error {
+
+	logger := p.logger.With(zap.Any("Client IP Address", client.RemoteAddr().String()), zap.Any("Client ConnectionID", ctx.Value(models.ClientConnectionIDKey).(string)), zap.Any("Destination ConnectionID", ctx.Value(models.DestConnectionIDKey).(string)))
+
+	//get the error group from the context
+	g, ok := ctx.Value(models.ErrGroupKey).(*errgroup.Group)
+	if !ok {
+		return errors.New("failed to get the error group from the context")
+	}
+
+	clientBuffChan := make(chan []byte)
+	destBuffChan := make(chan []byte)
+	errChan := make(chan error, 2)
+
+	// read requests from client
+	g.Go(func() error {
+		defer pUtil.Recover(logger, client, nil)
+		defer close(clientBuffChan)
+		pUtil.ReadBuffConn(ctx, logger, client, clientBuffChan, errChan)
+		return nil
+	})
+	// read responses from destination
+	g.Go(func() error {
```

Comment:
Can we make these method less repetitive? 

The entire block below is repeating: 
```
g.Go(func() error {
		defer pUtil.Recover(logger, client, nil)
		defer close(clientBuffChan)
		return nil
	})
```
Are these repeating elsewhere too?

Reply:
Yeah, we can make it less repetitive. This was referenced from the generic parser. I will make the changes in that file also.